tests/test_bqm.py

```diff
@@ -2312,3 +2312,49 @@ def test_quadratic_setitem(self, name, BQM):
         bqm.quadratic[('a', 'b')] = 5
         self.assertEqual(bqm.get_quadratic('a', 'b'), 5)
         assert_consistent_bqm(bqm)
+
+class TestViewsMinMax(unittest.TestCase):
+    @parameterized.expand([(cls.__name__, cls) for cls in BQM_SUBCLASSES])
+    def test_lin_minmax(self, name, BQM):
+        num_vars = 1000
+        D = np.arange(num_vars*num_vars).reshape((num_vars, num_vars))
+        bqm = BQM(D, 'SPIN') 
+
+        lmin = min(bqm.linear.values())
+        self.assertEqual(lmin, bqm.linear.min())
+
+        lmax = max(bqm.linear.values())
+        self.assertEqual(lmax, bqm.linear.max())
+
+    @parameterized.expand([(cls.__name__, cls) for cls in BQM_SUBCLASSES])
+    def test_quad_minmax(self, name, BQM):
+        num_vars = 1000
+        D = np.arange(num_vars*num_vars).reshape((num_vars, num_vars))
+        bqm = BQM(D, 'SPIN') 
+
+        qmin = min(bqm.quadratic.values())
+        self.assertEqual(qmin, bqm.quadratic.min())
+       
+        qmax = max(bqm.quadratic.values())
+        self.assertEqual(qmax, bqm.quadratic.max())
+
+    @parameterized.expand([(cls.__name__, cls) for cls in BQM_SUBCLASSES])
+    def test_lin_minmax_empty(self, name, BQM):
```

Comment:
I think this shouldn't be named `test_lin_minmax_empty` since the BQM has 500 variables? I would add some empty tests and make sure that they throw the correct error, something like
```
with self.assertRaises(ValueError):
    BQM('SPIN').linear.min()
self.assertEqual(BQM('SPIN').linear.min(10), 10)
```